### PR TITLE
cockpit-components-listing: React-related fixes

### PR DESCRIPTION
**Pull request by mareklibra**
Part of #9263 PR-split.

### CODE REVIEW

**Comment by martinpitt** [comment]
Fails to build with 

    make: *** No rule to make target 'dist/docker/Makefile.deps', needed by 'distdir'.  Stop.

**Comment by mareklibra** [comment]
Rebased

**Comment by martinpitt** [comment]
Still the same error. I thought it was just due to an old branch, but seems I looked at the wrong place in the log. This is the relevant one:

```
ERROR in ./pkg/lib/cockpit-components-listing.jsx

/tmp/source/pkg/lib/cockpit-components-listing.jsx
  229:77  error  No duplicate props allowed  react/jsx-no-duplicate-props
  231:77  error  No duplicate props allowed  react/jsx-no-duplicate-props
```

**Comment by mareklibra** [comment]
Fixed

File: pkg/lib/cockpit-components-listing.jsx [hunk 1]
```diff
@@ -304,34 +308,34 @@ var Listing = React.createClass({
             bodyClasses.push("listing-ct-wide");
         var headerClasses;
         var headerRow;
-        var selectableRows;
+        // var selectableRows;
         if (!this.props.children || this.props.children.length === 0) {
             headerClasses = "listing-ct-empty";
             headerRow = <tr><td>{this.props.emptyCaption}</td></tr>;
         } else if (this.props.columnTitles.length) {
+            /* probably not needed, ListingRow.selected prop should be set to 'false' by default
```
Comment (martinpitt) [inline pkg/lib/cockpit-components-listing.jsx]: What's your intention here? Let the tests run once, then delete this code? Or should this stay commented out? (subject to bitrot)? @mvollmer, WDYT?

Comment (mareklibra) [reply pkg/lib/cockpit-components-listing.jsx]: I intend to remove this. Reading through the code - I think it's not needed.
Tests passed on React, I expect same on react-lite but willing to see result first.
So putting needs-work back until it's clear.

Comment (mvollmer) [reply pkg/lib/cockpit-components-listing.jsx]: I have removed this commit from this PR.

**Pull request closed without merging**